indico/htdocs/js/indico/jquery/itempicker.js

```diff
@@ -0,0 +1,162 @@
+ * published by the Free Software Foundation; either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * Indico is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Indico; if not, see <http://www.gnu.org/licenses/>.
+ */
+
+(function($) {
+    $.widget('indico.itempicker', {
+
+        options: {
+            onSelect: null, // function called every time an item gets selected from the list
+            createItem: null, // function returning a jQuery object representing each item in the dropdown
+            items: null, // all the options that should be included in the dropdown
+            createCustomFooterElements: null,
+            containerClasses: '', // extra classes to be applied to the container
+            filterPlaceholder: '' // text displayed inside the input field
+        },
+
+        _create: function() {
+            var items = this.options.items || this.element.data('items') || [];
+            var qtipContent = this._createQtipContent(items);
+
+            this.element.qtip({
+                prerender: false,
+                overwrite: false,
+                style: {
+                    classes: 'item-picker-qtip ' + this.options.containerClasses
+                },
+                position: {
+                    my: 'right top',
+                    at: 'left top',
+                    target: this.element,
+                    adjust: {
+                        mouse: false,
+                        scroll: false
+                    }
+                },
+                content: {
+                    text: qtipContent
+                },
+                show: {
+                    event: 'click',
+                    solo: true
+                },
+                hide: {
+                    event: 'unfocus click'
+                },
+                events: {
+                    show: function() {
+                        _.delay(function() {
+                            qtipContent.find('.filter-input').focus();
+                        });
+                    }
+                }
+            });
+        },
+        _destroy: function() {
+            self.element.qtip('destroy');
+        },
+        _createQtipContent: function(items) {
+            var self = this;
+            var filterPlaceholder = self.options.filterPlaceholder || $T.gettext('Search');
+            var dropdownContainer = $('<div>', {'class': 'dropdown-container'});
+            var filterWrapper = $('<div>', {'class': 'dropdown-filter-wrapper'});
+            var filterInput = $('<input>', {'type': 'text', 'class': 'filter-input',
+                                            'css': {'height': 30, 'line-height': 30},
```

Comment:
Wait, `clearableinput` widget should be taking care of it. What is it that you are trying to do here?